src/Logger.cpp

```diff
@@ -86,3 +89,52 @@ void logger::error(const std::string format, ...) {
         std::flush(logFile);
     }
 }
+
+void logger::log_info(bool enable, const char *file, const char *function, const int line, const char *format, ... ) {
+    if (enable) {
+        char fileNonConst[256];
+        char message[256];
+        va_list ap;
+        strcpy(fileNonConst, file);
+        va_start(ap, format);
+        vsprintf(message, format, ap);

```

Comment:
Using `strcpy` and `vsprintf` is rather dangerous here: The input can be longer than the fixed buffer sizes, thus causing a stack overflow. `strncpy` and `vsnprintf` should be used, see `Logger::log` as an example for the latter.